src/main.ts

```diff
@@ -256,22 +258,55 @@ function handleKeyDown({ key }: KeyboardEvent): void {
 
 async function handleFindTrack(): Promise<void> {
   // Load track value
-  const trackId = trackInput.valueAsNumber;
+  await findTrack(trackInput.valueAsNumber);
+}
+
+async function findTrack(trackId: number): Promise<void> {
   const newTrack = dataset!.buildTrack(trackId);
 
-  if (newTrack.length() < 1) {  // Check track validity
+  if (newTrack.length() < 1) {
+    // Check track validity
     return;
   }
   selectedTrack = newTrack;
   await canv.setFrame(selectedTrack.times[0]);
   plot.plot(selectedTrack, featureName, canv.getCurrentFrame());
   await drawLoop();
+  trackInput.value = "" + trackId;
 }
 
 function resetTrackUI(): void {
   trackInput.value = "";
 }
 
+// URL STATE /////////////////////////////////////////////////////////////
```

Comment:
Because main is getting so big, this whole section could go into a new .ts file.. and that file or class could encapsulate both (1) getting datasetParam, trackParam, featureParam, and timeParam from the initial url, and (2) updating the current url when they change (even t... it's a bit odd for the url to change that fast...?).

Reply:
Actually there might be some better ways to do that, let me experiment and get back to you

Reply:
It's probably not really affecting performance. But this might be a good question to ask UX because simularium doesn't do it this way.  They only put t into a url when a user explicitly asks to generate a url for sharing via a command button.  The url is not expected to just hold all app state all the time.

Reply:
I suggest we leave it as is (the realtime updating) and see if it's bothering anyone or hurting playback perf.  Can fix it later

Reply:
unless you came up with a good way to handle it.  lol I got ahead of your comment about experimenting with it.

Reply:
Ah, no super elegant solutions unfortunately. I ended up implementing this anyways because I found that there was a noticeable lag if the URL was updated with every timestamp. It now updates when the pause button is pressed or whenever another feature (dataset, feature, track) changes.